### PR TITLE
ADEN-4015 Recover BFAB

### PR DESCRIPTION
**Pull request by fraszczakszymon**

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [helper.js#L16](https://github.com/Wikia/app/blob/6c8e40ec05892f4f502ac66e859b3647b8823ceb/extensions/wikia/AdEngine/js/provider/gpt/helper.js#L16): Function has 12 parameters which is greater than 7 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. MAJOR [helper.js#L51](https://github.com/Wikia/app/blob/6c8e40ec05892f4f502ac66e859b3647b8823ceb/extensions/wikia/AdEngine/js/provider/gpt/helper.js#L51): Function has a complexity of 14 which is greater than 10 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AFunctionComplexity)

File: extensions/wikia/AdEngine/js/provider/gpt/googleTag.js [hunk 1]
```diff
@@ -22,14 +22,20 @@ define('ext.wikia.adEngine.provider.gpt.googleTag', [
...
-
 		for (id in registeredCallbacks) {
-			if (registeredCallbacks.hasOwnProperty(id) &&
-				registeredCallbacks[id] && event.slot && event.slot === googleSlots.getSlot(id)
-			) {
-				log(['dispatchEvent', event, id, 'Launching registered callback'], 'debug', logGroup);
-				registeredCallbacks[id](event);
-				return;
+			if (registeredCallbacks.hasOwnProperty(id)) {
+				var slotId = id;
```
Comment (v3rth) [inline extensions/wikia/AdEngine/js/provider/gpt/googleTag.js]: You need `slotId` only for `googleSlots.getSlot(slotId)`, maybe close this abstraction behind some additional function

File: extensions/wikia/AdEngine/js/provider/gpt/googleTag.js [hunk 2]
```diff
@@ -22,14 +22,20 @@ define('ext.wikia.adEngine.provider.gpt.googleTag', [
...
+
+				if (recovery.isRecoveryEnabled() && recovery.isBlocking()) {
+					slotId = window._sp_.getElementId(slotId);
+				}
+
+				if (registeredCallbacks[id] && event.slot && event.slot === googleSlots.getSlot(slotId)) {
```
Comment (v3rth) [inline extensions/wikia/AdEngine/js/provider/gpt/googleTag.js]: Extract condition to some function it's hard to understand what and for what it happen

File: extensions/wikia/AdEngine/js/provider/gpt/googleTag.js [hunk 3]
```diff
@@ -22,14 +22,20 @@ define('ext.wikia.adEngine.provider.gpt.googleTag', [
...
-			if (registeredCallbacks.hasOwnProperty(id) &&
-				registeredCallbacks[id] && event.slot && event.slot === googleSlots.getSlot(id)
-			) {
-				log(['dispatchEvent', event, id, 'Launching registered callback'], 'debug', logGroup);
-				registeredCallbacks[id](event);
-				return;
+			if (registeredCallbacks.hasOwnProperty(id)) {
+				var slotId = id;
+
+				if (recovery.isRecoveryEnabled() && recovery.isBlocking()) {
```
Comment (drets) [inline extensions/wikia/AdEngine/js/provider/gpt/googleTag.js]: I thought about adding unit tests for `dispactEvent` function, but it looks like it requires a lot of tweaks. :confused:

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [helper.js#L16](https://github.com/Wikia/app/blob/a326a69365962ac0cd7d717cc5e0fa5678c61d28/extensions/wikia/AdEngine/js/provider/gpt/helper.js#L16): Function has 12 parameters which is greater than 7 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. MAJOR [helper.js#L51](https://github.com/Wikia/app/blob/a326a69365962ac0cd7d717cc5e0fa5678c61d28/extensions/wikia/AdEngine/js/provider/gpt/helper.js#L51): Function has a complexity of 14 which is greater than 10 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AFunctionComplexity)

**Review by drets — APPROVED** [review]

File: extensions/wikia/AdEngine/js/template/bfab.js [hunk 4]
```diff
@@ -1,18 +1,53 @@
...
+		if (isRecoveryNotSupportedBrowser()) {
+			DOMElementTweaker.hide(adContainer, true);
+			return;
+		}
+
+		slotTweaker.tweakRecoveredSlot(originalIframe, iframe);
+	}
+
 	function show(params) {
+		console.log(params);
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/template/bfab.js]: MAJOR Remove this logging statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS2228)

File: extensions/wikia/ARecoveryEngine/js/helper.js [hunk 5]
```diff
@@ -49,6 +51,34 @@ define('ext.wikia.aRecoveryEngine.recovery.helper', [
...
+
+		div.appendChild(originalIframe);
+		parentNode.appendChild(div);
+		originalIframe.style.display = 'none';
+		div.style.display = 'none';
+
+		return originalIframe;
+	}
+
+	function recoverSlot(iframe, params) {
```
Comment (WikiaTech) [inline extensions/wikia/ARecoveryEngine/js/helper.js]: MAJOR Remove unused function 'recoverSlot'. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AUnusedVariable)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 5 issues
* MAJOR 5 major

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [helper.js#L98](https://github.com/Wikia/app/blob/26efd039aefb1c8ce06d0e8129dac20eacfdf6f2/extensions/wikia/ARecoveryEngine/js/helper.js#L98): Either remove or fill this block of code. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AEmptyBlock)
1. MAJOR [helper.js#L16](https://github.com/Wikia/app/blob/26efd039aefb1c8ce06d0e8129dac20eacfdf6f2/extensions/wikia/AdEngine/js/provider/gpt/helper.js#L16): Function has 12 parameters which is greater than 7 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. MAJOR [helper.js#L51](https://github.com/Wikia/app/blob/26efd039aefb1c8ce06d0e8129dac20eacfdf6f2/extensions/wikia/AdEngine/js/provider/gpt/helper.js#L51): Function has a complexity of 14 which is greater than 10 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AFunctionComplexity)

File: extensions/wikia/AdEngine/js/template/bfaa.js [hunk 6]
```diff
@@ -3,28 +3,24 @@ define('ext.wikia.adEngine.template.bfaa', [
...
 	'ext.wikia.adEngine.slotTweaker',
-	'ext.wikia.aRecoveryEngine.recovery.helper',
-	'wikia.browserDetect',
 	'wikia.document',
 	'wikia.log',
 	'wikia.window',
-	require.optional('ext.wikia.adEngine.mobile.mercuryListener')
+	require.optional('ext.wikia.adEngine.mobile.mercuryListener'),
+	require.optional('ext.wikia.aRecoveryEngine.recovery.tweaker')
 ], function (
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/template/bfaa.js:13]: MAJOR Function has 10 parameters which is greater than 7 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AExcessiveParameterList)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 6 issues
* CRITICAL 1 critical
* MAJOR 3 major
* INFO 2 info

Watch the comments in this conversation to review them.

#### 5 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. CRITICAL [SlotTweaker.js#L206](https://github.com/Wikia/app/blob/6f5fceb312d48776cb7b1d2b9c80b78ea9951486/extensions/wikia/AdEngine/js/SlotTweaker.js#L206): "noop" expects "0" arguments, but "1" were provided. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS930)
1. MAJOR [helper.js#L16](https://github.com/Wikia/app/blob/6f5fceb312d48776cb7b1d2b9c80b78ea9951486/extensions/wikia/AdEngine/js/provider/gpt/helper.js#L16): Function has 12 parameters which is greater than 7 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. MAJOR [helper.js#L51](https://github.com/Wikia/app/blob/6f5fceb312d48776cb7b1d2b9c80b78ea9951486/extensions/wikia/AdEngine/js/provider/gpt/helper.js#L51): Function has a complexity of 14 which is greater than 10 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AFunctionComplexity)
1. INFO [SlotTweaker.js#L72](https://github.com/Wikia/app/blob/6f5fceb312d48776cb7b1d2b9c80b78ea9951486/extensions/wikia/AdEngine/js/SlotTweaker.js#L72): Complete the task associated to this TODO comment. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS1135)
1. INFO [SlotTweaker.js#L80](https://github.com/Wikia/app/blob/6f5fceb312d48776cb7b1d2b9c80b78ea9951486/extensions/wikia/AdEngine/js/SlotTweaker.js#L80): Complete the task associated to this TODO comment. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS1135)

File: extensions/wikia/ARecoveryEngine/js/recovery/tweaker.js [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+	recoveryHelper,
+	browser,
+	doc,
+	log
+) {
+	'use strict';
+
+	var logGroup = 'ext.wikia.aRecoveryEngine.recovery.tweaker';
+
+	function isSupportedBrowser() {
```
Comment (bognix) [inline extensions/wikia/ARecoveryEngine/js/recovery/tweaker.js]: so IE and Edge are in fact not supported browsers?
(https://github.com/Wikia/app/pull/11625/files#diff-22ed2cfab091f6df9365608f88b88eb8R43)

**Review by bognix — APPROVED** [review]
Rename the method and LGTM

**Comment by WikiaTech** [comment]
SonarQube analysis reported 6 issues
- CRITICAL 1 critical
- MAJOR 3 major
- INFO 2 info

Watch the comments in this conversation to review them.

#### 5 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. CRITICAL [SlotTweaker.js#L206](https://github.com/Wikia/app/blob/0f9eea5b2ae39fb27d49713e198d1c2ef94f2a77/extensions/wikia/AdEngine/js/SlotTweaker.js#L206): "noop" expects "0" arguments, but "1" were provided. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS930)
2. MAJOR [helper.js#L16](https://github.com/Wikia/app/blob/0f9eea5b2ae39fb27d49713e198d1c2ef94f2a77/extensions/wikia/AdEngine/js/provider/gpt/helper.js#L16): Function has 12 parameters which is greater than 7 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AExcessiveParameterList)
3. MAJOR [helper.js#L51](https://github.com/Wikia/app/blob/0f9eea5b2ae39fb27d49713e198d1c2ef94f2a77/extensions/wikia/AdEngine/js/provider/gpt/helper.js#L51): Function has a complexity of 14 which is greater than 10 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AFunctionComplexity)
4. INFO [SlotTweaker.js#L72](https://github.com/Wikia/app/blob/0f9eea5b2ae39fb27d49713e198d1c2ef94f2a77/extensions/wikia/AdEngine/js/SlotTweaker.js#L72): Complete the task associated to this TODO comment. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS1135)
5. INFO [SlotTweaker.js#L80](https://github.com/Wikia/app/blob/0f9eea5b2ae39fb27d49713e198d1c2ef94f2a77/extensions/wikia/AdEngine/js/SlotTweaker.js#L80): Complete the task associated to this TODO comment. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS1135)

**Pull request merged**